spras/analysis/ml.py

```diff
@@ -86,18 +86,17 @@ def summarize_networks(file_paths: Iterable[Union[str, PathLike]]) -> pd.DataFra
     concated_df = concated_df.fillna(0)
     concated_df = concated_df.astype('int64')
 
-    # don't do ml post-processing if there is an empty dataframe or the number of samples is <= 1
-    if concated_df.empty:
+    return concated_df
+
+def df_error(dataframe: pd.DataFrame):
```

Comment:
an alternative approach to this could be creating a custom exception class for validation errors and having seperate methods that check for empty dfs/incorrect shapes? could help with modularity (but could also be overkill?)

Reply:
Now that I've read all the code, I think it is okay to keep this as a function. However, I would rename it to `validate_df` or something similar to indicate its role is to help with input validation. It isn't a custom error.